python_ta/contracts/__init__.py

```diff
@@ -321,17 +331,71 @@ def _check_function_contracts(wrapped, instance, args, kwargs):
 
 
 def check_type_strict(argname: str, value: Any, expected_type: type) -> None:
-    """Ensure that ``value`` matches ``expected_type``.
+    """
+    Ensure that `value` matches ``expected_type`` with strict type checking.
 
-    Differentiates between:
-        - float vs. int
-        - bool vs. int
+    This function enforces strict type distinctions within the numeric hierarchy (bool, int, float,
+    complex), ensuring that the type of value is exactly the same as expected_type, not merely a subtype.
```

Comment:
Remove the phrase "not merely a subtype". Technically, `int` isn't a subclass of `float` (though `bool` _is_ a subclass of `int`), it's just that PEP 484 says type checkers can treat them as such.